app/settings/server/functions/settings.ts

```diff
@@ -207,6 +207,19 @@ class Settings extends SettingsBase {
 
 		updateOperations.$set.ts = new Date();
 
+		const record = {
+			_id,
+			value,
+			type: options.type || 'string',
+			env: options.env || false,
+			i18nLabel: options.i18nLabel,
+			public: options.public || false,
+			packageValue: options.packageValue,
+			blocked: options.blocked,
+		};
+
+		this.storeSettingValue(record, this.initialLoad);
```

Comment:
Should this happen after the upsert?  That way if has issue storing to db the cache won't be updated

Reply:
Indeed. I've updated the PR to store the value after the db upsert.